swarm/api/api.go

```diff
@@ -691,39 +688,34 @@ func (a *API) ResourceCreate(ctx context.Context, name string, frequency uint64)
 
 // ResourceUpdateMultihash updates a Mutable Resource and marks the update's content to be of multihash type, which will be recognized upon retrieval.
 // It will fail if the data is not a valid multihash.
-func (a *API) ResourceUpdateMultihash(ctx context.Context, name string, data []byte) (storage.Address, uint32, uint32, error) {
-	return a.resourceUpdate(ctx, name, data, true)
+func (a *API) ResourceUpdateMultihash(ctx context.Context, addr storage.Address, data []byte) (storage.Address, uint32, uint32, error) {
+	return a.resourceUpdate(ctx, addr, data, true)
 }
 
 // ResourceUpdate updates a Mutable Resource with arbitrary data.
 // Upon retrieval the update will be retrieved verbatim as bytes.
-func (a *API) ResourceUpdate(ctx context.Context, name string, data []byte) (storage.Address, uint32, uint32, error) {
-	return a.resourceUpdate(ctx, name, data, false)
+func (a *API) ResourceUpdate(ctx context.Context, addr storage.Address, data []byte) (storage.Address, uint32, uint32, error) {
+	return a.resourceUpdate(ctx, addr, data, false)
 }
 
-func (a *API) resourceUpdate(ctx context.Context, name string, data []byte, multihash bool) (storage.Address, uint32, uint32, error) {
-	var addr storage.Address
+func (a *API) resourceUpdate(ctx context.Context, rootAddr storage.Address, data []byte, multihash bool) (storage.Address, uint32, uint32, error) {
+	var updateAddr storage.Address
 	var err error
 	if multihash {
-		addr, err = a.resource.UpdateMultihash(ctx, name, data)
+		updateAddr, err = a.resource.UpdateMultihash(ctx, rootAddr, data)
 	} else {
-		addr, err = a.resource.Update(ctx, name, data)
+		updateAddr, err = a.resource.Update(ctx, rootAddr, data)
 	}
-	period, _ := a.resource.GetLastPeriod(name)
-	version, _ := a.resource.GetVersion(name)
-	return addr, period, version, err
+	period, _ := a.resource.GetLastPeriod(rootAddr)
```

Comment:
I don't mind fixing this in #704 so I can avoid a potential conflict. Please let me know.
In #704, I would remove:
```
	period, _ := a.resource.GetLastPeriod(rootAddr)
	version, _ := a.resource.GetVersion(rootAddr)
```
... since `period`, `version` are a given and not a result of calling `Update` and they are part of the structure that gets passed in anyway, so you would only care about the error result of `Update`

Reply:
cool

Reply:
Confirmed this is gone in #704.